test/tests/functional/pbs_test_tpp.py

```diff
@@ -381,29 +380,30 @@ def test_multiple_comm_with_mom(self):
         This test verifies communication between server-mom and
         between mom when multiple pbs_comm are present in cluster
         Configuration:
-        Node 1 : Server, Sched, Comm
+        Node 1 : Server, Sched, Mom, Comm
         Node 2 : Mom
-        Node 3 : Mom
-        Node 4 : Comm
+        Node 3 : Comm
         """
         self.momA = self.moms.values()[0]
         self.momB = self.moms.values()[1]
-        self.comm1 = self.comms.values()[0]
         self.comm2 = self.comms.values()[1]
+        if self.moms.values()[0].shortname == self.server.shortname:
+            self.momA = self.moms.values()[0]
+            self.momB = self.moms.values()[1]
+        else:
+            self.momA = self.moms.values()[1]
+            self.momB = self.moms.values()[0]
+
         self.hostA = self.momA.shortname
         self.hostB = self.momB.shortname
-        self.hostC = self.comm1.shortname
-        self.hostD = self.comm2.shortname
-        nodes = [self.hostA, self.hostB, self.hostC, self.hostD]
+        self.comm2 = self.comms.values()[1]
```

Comment:
I think this would pass if "-p comms=m1:m3" is passed and would fail if "-p comms=m3,m1" is passed. Hence, needs update to not fail.